pkg/resource/deploy/step_generator.go

```diff
@@ -623,6 +623,16 @@ func (sg *stepGenerator) generateSteps(event RegisterResourceEvent) ([]Step, err
 		contract.Assertf(n == len(randomSeed),
 			"generated fewer (%d) than expected (%d) random bytes", n, len(randomSeed))
 	}
+	var autonaming *plugin.AutonamingOptions
+	var autonamingDeleteBeforeCreate bool
+	if sg.deployment.opts.Autonaming != nil {
+		a, dbc, err := sg.deployment.opts.Autonaming.AutonamingForResource(urn, randomSeed)
```

Comment:
If you make this `=` instead of `:=` I think you save the ugly assignments below. So `autonaming, autonamingDeleteBeforeReplace, err = ...`. You might need `var err error` if it's new but still neater/less error prone I think?